.github/workflows/sonar.yml

```
@@ -0,0 +1,127 @@
+        run: echo "$GITHUB_CONTEXT"
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+
+  analyze:
+    if: ${{ github.repository == 'bf2fc6cc711aee1a0c2a/kas-fleetshard' && github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ubuntu-latest
+    name: Analyze
+    steps:
+      - name: Display Github Event Context
+        run: echo "$GITHUB_CONTEXT"
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+
+      ## Checkout the source of the event that triggered this workflow,
+      ## PR commit (pull_request event) or commit (push event).
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ github.event.workflow_run.head_repository.full_name }}
+          ref: ${{ github.event.workflow_run.head_sha }}
+          fetch-depth: 0
+
+      ## Retrieve the `target` directory from the build job
+      - name: Fetch Build Result
+        uses: actions/github-script@v3.1.0
+        with:
+          script: |
+            var artifacts = await github.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: ${{ github.event.workflow_run.id }},
+            });
+            var matchArtifact = artifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "target"
+            })[0];
+            var download = await github.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            var fs = require('fs');
+            fs.writeFileSync('${{github.workspace}}/target.zip', Buffer.from(download.data));
+      ## Extract the `target` directory from the build job
+      - name: Extract Build Result
+        run: |
+          unzip target.zip
+      ## Load the context from the build job - runs for any trigger to allow templates with `steps.build_context.outputs.content`
+      ## to be accepted by GitHub Actions.
+      - name: Read Build Context
+        id: build_context
+        uses: juliangruber/read-file-action@v1
+        with:
+          path: ./target/build-context.json
+
+      ## (PRs Only) Check out the base branch (target of the PR)
+      - name: Checkout Base Branch (PR Only)
+        if: github.event.workflow_run.event == 'pull_request'
+        env:
+          BASE_BRANCH: ${{ fromJson(steps.build_context.outputs.content).base_ref }}
+        run: |
+          git remote add upstream ${{ github.event.repository.clone_url }}
+          git fetch upstream --prune --tags --force
+          git checkout -B $BASE_BRANCH upstream/$BASE_BRANCH
+          git checkout ${{ github.event.workflow_run.head_sha }}
+          git clean -ffdx --exclude=target/ && git reset --hard HEAD
+      - name: Setup JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 17
```

Use `11` for kas-fleetshard
```suggestion
          java-version: 11
```